test/collection/multimap_test.dart

```diff
@@ -64,6 +64,41 @@ void main() {
     }, tags: ['fails-on-dartdevc']);
   });
 
+  group('Multimap.fromMultimap', () {
+    late Multimap<String, String> mmap1;
+    late Multimap<String, String> mmap2;
+    setUp(() {
+      mmap1 = Multimap()
+        ..add('k1', 'v1')
+        ..add('k1', 'v2')
+        ..add('k2', 'v3');
+
+      mmap2 = Multimap.fromMultimap(mmap1);
+    });
+
+    test('should have different memory references', () {
+      expect(false, mmap1 == mmap2);
+    });
+
+    test('should have same keys', () {
+      var map1ContainsAllMap2Keys =
+          mmap1.keys.every((key) => mmap2.containsKey(key));
+      var map2ContainsAllMap1Keys =
+          mmap2.keys.every((key) => mmap1.containsKey(key));
+
+      expect(true, map1ContainsAllMap2Keys && map2ContainsAllMap1Keys);
+    });
+
+    test('should have same values', () {
+      var map1ContainsAllMap2Values =
+          mmap1.values.every((value) => mmap2.containsValue(value));
+      var map2ContainsAllMap1Values =
+          mmap2.values.every((value) => mmap1.containsValue(value));
+
+      expect(true, map1ContainsAllMap2Values && map2ContainsAllMap1Values);
+    });
+  });
```

Comment:
We should add a test to ensure no one inadvertently breaks the behaviour by, say, re-using the Lists/Sets we create internally in mmap1 -- either by modifying the factory constructor or by modifying addAll. I haven't looked to see that we already have this test for addAll (hopefully), but either way, it's worth being paranoid here.

Something like:
```dart
mmap1 = Multimap()
  ..add('k1', 'v1')
  ..add('k1', 'v2')
  ..add('k2', 'v3');
mmap2 = Multimap.fromMultimap(mmap1);
mmap1['k1'].remove('v1');
expect(mmap2['k1'].contains('v1'));
```